test/parallel/test-stringbytes-external.js

```diff
@@ -119,3 +122,53 @@ var PRE_3OF4_APEX = Math.ceil((EXTERN_APEX / 4) * 3) - RADIOS;
   assert.equal(a, b);
   assert.equal(b, c);
 })();
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  var kStringMaxLength = internalBuffer.kStringMaxLength;
+
+  var maxString = new Buffer(kStringMaxLength).toString();
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength).toString('binary');
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength + 1).toString('binary', 1);
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength + 1).toString(
+      'binary', 0, kStringMaxLength);
```

Comment:
Pico-style nit: the convention is to put the `new Buffer(...` on a new line, indented by four spaces (although the linter probably accepts it as is.)